### DISCUSSION TITLE
Add Element

### DISCUSSION THREAD

**Discussion by Yueqi77-amazing** [mjhoptics/ray-optics#94] [Q&A] [answered]
Hello Mr.Hayford,
If i want to add a cemented Element like the AddFromFileAndFlip.rst showed, but i want to add it like in the sequence model using sm.add_surface() function, is there a function to make it happens since i only found in example with add_from_file function which i don’t want to use.
Thanks 
Yvonne

**Comment by mjhoptics** [comment, marked as answer]
Hello Yvonne,
ray-optics should be able to detect when a cemented doublet can be used based on the sequential model. For instance:

```
from rayoptics.environment import *
opm = OpticalModel()
sm  = opm['seq_model']
osp = opm['optical_spec']

osp['pupil'] = PupilSpec(osp, key=['object', 'pupil'], value=22.)
osp['fov'] = FieldSpec(osp, key=['object', 'angle'], value=2.0, flds=[0., 1.], is_relative=True)
osp['wvls'] = WvlSpec([('d', 1.0)], ref_wl=0)

opm.radius_mode = False

sm.gaps[0].thi=1e10

sm.add_surface([0.016268, 6.00000 , 'N-BK7', 'Schott', 12.000])
sm.add_surface([-0.022401, 2.50000, 'N-SF5', 'Schott', 12.289])
sm.add_surface([-0.007696, 10.0000])

# this will recognize and create a CementedElement
opm.update_model()

opm['part_tree'].list_model()

layout_plt = plt.figure(FigureClass=InteractiveLayout, opt_model=opm).plot()
```

produces

```
root
├── Object
├── CE1
└── Image
```
![Figure 2](https://user-images.githubusercontent.com/33294031/185555375-80316919-5d94-448c-af42-6bf7ec7abc64.png)

Hope this helps.

Mike

**Reply by Yueqi77-amazing** [reply]
Thanks for your answer!
